pkg/plugin/plugin_test.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"testing"
+
+	g "github.com/onsi/ginkgo" // An alias is required because Context is defined elsewhere in this package.
+	. "github.com/onsi/gomega"
+)
+
+func TestCLI(t *testing.T) {
+	RegisterFailHandler(g.Fail)
+	g.RunSpecs(t, "Plugin Suite")
+}
+
+var _ = g.Describe("ParseVersion", func() {
+
+	var (
+		v   Version
+		err error
+	)
+
+	g.It("should pass validation", func() {
+		g.By("validating 1")
+		v, err = ParseVersion("1")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(v.Number).To(BeNumerically("==", int64(1)))
+		Expect(v.Stability).To(Equal(""))
+
```

Comment:
I still not happy with the tests because still not possible read them t not what/when/why as its docs examples.
Let's try to clarify. See that they are : 
- should pass validation by validating 1 - Why? 
- should pass validation by validating 2 - Why? 
- should pass validation by validating 22 - Why? 

When I think that they should be: 

```go
Describe("Checking plugin version validation", func() {

It("should work successfully when a plugin version is a positive number")
By(passing version number as 1)
By(passing version number as 2)

It("should fail when a plugin version is a negative number")
By(passing version number as -2)

It("should work successfully when a plugin version has a stage nomenclature)
By(passing version stage as alpha)
By(passing version stage as beta)

It(should fail if the plugin version has an invalid stage nomenclature)
```

Reply:
Done.